src/components/typography/Typography.tsx

```diff
@@ -0,0 +1,49 @@
+// src/components/Typography/Typography.tsx
+
+import React from 'react';
+import { Text, TextProps } from 'react-native';
+
+import { styles } from './Typography.styles';
+
+interface TypographyProps extends TextProps {
+  variant?: 'h1' | 'h2' | 'h3' | 'body' | 'caption';
+  color?: string;
+  fontSize?: number;
+  fontWeight?: 'normal' | 'bold' | number;
+  textAlign?: 'left' | 'right' | 'center' | 'justify';
+  children: React.ReactNode;
+}
+
+const Typography: React.FC<TypographyProps> = ({
+  variant,
+  color,
+  fontSize,
+  fontWeight,
+  textAlign,
+  style,
+  children,
+  ...rest
+}) => {
+  return (
+    <Text
+      style={[
+        styles[variant], // Apply variant styles from Typography.styles.ts
```

Comment:
remove comments, see [How to write better code comments](https://dev.to/adammc331/todo-write-a-better-comment-4c8c)